QueueCustomImplementation/Queue.cs

```diff
@@ -0,0 +1,92 @@
+﻿using QueueCustomImplementation.Interfaces;
+
+namespace QueueCustomImplementation
+{
+    public class Queue<T> : IGenericQueue<T>
+    {
+        private int _elementCount;
+        private int _maxCapacity;
+        private T[] _elements;
+
+        public Queue(int queueElementsCapacity)
+        {
+            _maxCapacity = queueElementsCapacity;
+            _elements = new T[_maxCapacity];
+            _elementCount = 0;
+        }
+
+        public T Dequeue()
+        {
+            var firstItem = _elements[0];
+
+            Drop();
+
+            return firstItem;
+        }
+
+        public void Drop()
+        {
+            if (IsEmpty())
+            {
+                throw new InvalidOperationException("Queue is empty.");
+            }
+
+            var newElements = new T[_elementCount - 1];
```

Comment:
Think how to implement functionality not creating new array to improve performance (hint- use array  _maxCapacity *2 and change the pointer to first element not copying elements)

Reply:
Fixed